src/aggregation/agg_result.rs

```diff
@@ -81,12 +82,18 @@ impl From<IntermediateMetricResult> for MetricResult {
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
 #[serde(untagged)]
 pub enum BucketResult {
-    /// This is the default entry for a bucket, which contains a key, count, and optionally
+    /// This is the range entry for a bucket, which contains a key, count, from, to, and optionally
     /// sub_aggregations.
     Range {
         /// The range buckets sorted by range.
         buckets: Vec<RangeBucketEntry>,
     },
+    /// This is the histogram entry for a bucket, which contains a key, count, and optionally
+    /// sub_aggregations.
+    Histogram {
+        /// The buckets.
+        buckets: Vec<BucketEntry>,
```

Comment:
I don't understand. Can key be anything?
Can there be holes?

Reply:
BucketEntry is a general purpose bucket result, because many bucket aggregations have `key`, `doc_count` and `sub_aggregations`.

Key is in this case always Key::F64. Do you think a specialized version would be better?

If there are holes depends on the request, if min_doc_count is 0, then there are no holes

Reply:
Why is it different from `RangeBucketEntry`? I thought histogram was like range but with a shorter API and the opportunity for optimization.

Reply:
`RangeBucketEntry` contains the from and to values, histogram has just the f64 key (the min value of the bucket)

> Can we put some comment in there about the presence of holes ?
Yep I'll add it

Reply:
Yes but why? Is it for serialization purpose?